x-pack/plugins/apm/public/components/shared/unified_search_bar/index.tsx

```diff
@@ -148,8 +172,23 @@ export function UnifiedSearchBar({
     }
     // Sync Time Range with Search Bar
     timeFilterService.timefilter.setTime(urlTimeRange as TimeRange);
-  }, [kuery, queryStringService, timeFilterService, urlTimeRange]);
+  }, [kuery, queryStringService, timeFilterService.timefilter, urlTimeRange]);
 
+  useEffectOnce(() => {
+    // Sync Refresh Interval with Search Bar
+    const refreshInterval = {
+      pause: toBoolean(refreshPausedFromUrl),
+      value: refreshIntervalFromUrl,
+    };
+    if (
+      !deepEqual(
+        timeFilterService.timefilter.getRefreshInterval(),
+        refreshInterval
+      )
```

Comment:
why do we need the (deep) equal check? what happens if it's the same value and we call `setRefreshInterval` anyway? will it cause infinite recursion?

Reply:
Yup can be removed